airflow/models/taskinstance.py

```diff
@@ -1651,16 +1646,23 @@ def signal_handler(signum, frame):
                 )
 
             # Run pre_execute callback
-            self.task.pre_execute(context=context)
+            # Is never MappedOperator at this point
+            self.task.pre_execute(context=context)  # type: ignore[union-attr]
```

Comment:
Where is the task unmapped before this line is reached? How can we be sure this is the case?

Reply:
@uranusjr I'm actually going by your code 😉 

https://github.com/apache/airflow/commit/5560a46bfe8a14205c5e8a14f0b5c2ae74ee100c

Reply:
The fact I wrote the code is not a particularly promising guarantee though 😛 

But code here is pretty awful. I’ll add a commit to better fix this up later…